app/assets/javascripts/galleries/uploader.js

```diff
@@ -1,26 +1,26 @@
-/* global addUploadedIcon */
-$(document).ready(function() {
-  bindFileInput($("#icon_files"));
-});
+/* global addUploadedIcon, originalUrl, updateBox, total, failed, done */
 
-function updateBox(progressBox, done, total, failed) {
-  if (!progressBox) return;
-  var progress = parseInt(done / total * 100, 10);
-  progressBox.html(done.toString() + ' / ' + total.toString() + ' (' + progress + '%) ');
-  if (failed) {
-    progressBox.append($("<span style='color: #f00;'>").append(failed.toString() + " failed"));
-  }
-}
+var uploadedIcons = {};
 
-function bindFileInput(fileInput) {
+$(document).ready(function() {
   var form = $('form.icon-upload');
   var submitButton = form.find('input[type="submit"]');
   var formData = form.data('form-data');
-  var progressBox = fileInput.closest('td').find('.progress-box');
-  var done = 0;
-  var total = 0;
-  var failed = 0;
 
+  $.each($(".icon_files"), function( index, fileInput ) {
```

Comment:
Not that it's important but I think most of these `$.each($(selector), function(…){…})` (and same for `$.map`) can be done in the format `$(selector).each(function(…){…})`